pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -173,12 +174,19 @@ func (gb *GraphBuilder) controllerFor(logger klog.Logger, resource schema.GroupV
 	shared, err := gb.sharedInformers.ForResource(resource)
 	if err != nil {
 		logger.V(4).Error(err, "unable to use a shared informer", "resource", resource, "kind", kind)
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	logger.V(4).Info("using a shared informer", "resource", resource, "kind", kind)
 	// need to clone because it's from a shared cache
-	shared.Informer().AddEventHandlerWithResyncPeriod(handlers, ResourceResyncTime)
-	return shared.Informer().GetController(), shared.Informer().GetStore(), nil
+	registration, err := shared.Informer().AddEventHandlerWithResyncPeriod(handlers, ResourceResyncTime)
```

Comment:
This removes the event handler but doesn't do anything to clear out the things observed from this API from the GC graph

There are two cases I think we need to handle differently:
* Removal of a CRD: if GC is watching widgets v1beta1, and I delete the widgets v1beta1 API, widget instances should probably get cleared from the graph?
* Addition of a new version to a CRD: if GC is watching widgets v1beta1, and I update the CRD to define a v1 version, that will show up as "start watching v1, stop watching v1beta1". In that case, I think we would want to transition instances in the GC graph to the v1 API, not remove and re-add them.

We need test coverage for both of those cases to make sure we do the right thing.